oak_functions/loader/src/server.rs

```diff
@@ -649,19 +672,25 @@ impl WasmHandler {
         let mut extensions_indices = HashMap::new();
         let mut extensions_metadata = HashMap::new();
         // Remove the extension_endpoints map as soon as we extend extensions by functionality for
-        // channels. Until then we use the (unique) ABI_FUNCTION_NAME as keys.
+        // channels. Until then we use the ChannelHandle as keys.
         let mut extensions_endpoints = HashMap::new();
 
         let mut channel_switchboard = ChannelSwitchboard::new();
-        // Register only lookup channel for now.
-        let lookup_endpoint = channel_switchboard.register(ChannelHandle::LookupData);
-        extensions_endpoints.insert(LOOKUP_ABI_FUNCTION_NAME.to_string(), lookup_endpoint);
 
         for (ind, factory) in self.extension_factories.iter().enumerate() {
             let extension = factory.create()?;
-            let (name, signature) = extension.get_metadata();
-            extensions_indices.insert(ind + EXTENSION_INDEX_OFFSET, extension);
-            extensions_metadata.insert(name, (ind + EXTENSION_INDEX_OFFSET, signature));
+            match extension {
+                Extension::Native(ref native_extension) => {
+                    let (name, signature) = native_extension.get_metadata();
+                    extensions_indices.insert(ind + EXTENSION_INDEX_OFFSET, extension);
+                    extensions_metadata.insert(name, (ind + EXTENSION_INDEX_OFFSET, signature));
+                }
+                Extension::Uwabi(uwabi_extension) => {
+                    let channel_handle = uwabi_extension.get_channel_handle();
+                    let endpoint = channel_switchboard.register(channel_handle);
+                    extensions_endpoints.insert(channel_handle, endpoint);
```

Comment:
Looks like <`extensions_indices`, `extensions_metadata`> and <`channel_switchboard`, `extensions_endpoints`> are somewhat mutually exclusive now. Is there a way to unify these constructs? Or perhaps rename them to clearly reflect the connection between them? How is `extensions_endpoints` related to `extensions_indices` and `extensions_metadata`?

Reply:
Yes, you are right---they are mutually exclusive. The `extension_indices` and `extension_metadata` is related to the native extensions and the `channel_switchboard` and `extension_endpoints` to the Uwabi extensions.

I actually want to remove the `extension_endpoints` in the next step (#2502), so I would revisit that then. 

WDYT?

Reply:
Good idea: #2505 .